spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/ParameterMetadataProvider.java

```diff
@@ -241,7 +243,7 @@ public Object prepare(Object value) {
 				return value;
 			}
 
-			if (String.class.equals(expressionType)) {
+			if (String.class.equals(expressionType) && !noWildcards) {
```

Comment:
Better to create a separate method for this
```java
private boolean aMethod() {
    return String.class.equals(expressionType) && !noWildcards;
}
```
 and say in the method name what is actually checked here.